### DISCUSSION TITLE
send data more than 64 bytes in HID Communication

### DISCUSSION THREAD

**Discussion by nachiketathakur** [hathach/tinyusb#1447] [Q&A] [answered]
In my application on RP2040, I want to send data from device to HOST of size more than 64 bytes(800 bytes). But I cannot figure out how to do that. When the size of the macro CFG_TUD_HID_EP_BUFSIZE to 800, it is not working. I am using in Full Speed mode. Can anyone suggest me something? 
Thanks.

**Comment by hathach** [comment, marked as answer]
for HID, you need to send 64-byte packet one after another. you could use the complete callback to help with back-to-back send

https://github.com/hathach/tinyusb/blob/6ec5174d8b24d1f01a443c74a0a3dbee00523efc/src/class/hid/hid_device.h#L119

**Reply by nachiketathakur** [reply]
Thank you for the response. Is there an example code which uses this function?

**Reply by hathach** [reply]
maybe there is some, you could try to search  for the function `tud_hid_report_complete_cb()` in example/google/github to see if there is others using it.
